src/reducers/selectors/getCdpData.js

```diff
@@ -0,0 +1,83 @@
+import * as math from '@makerdao/dai-plugin-mcd/dist/math';
+import { Currency } from '@makerdao/currency';
+import ilkList from '../../references/ilkList';
+import mapValues from 'lodash/mapValues';
+
+export default function getCdpData(id, store, formatted) {
```

Comment:
this selector is the crux of the new structure -- we store mostly raw data for urns and ilks in the store, and calculate the necessary human-readable values here, optionally formatting them for rendering. if anything here turns out to be expensive enough to cache, then we can do so, at the cost of having to track dependencies from derived to raw values.

Reply:
Maybe observables would be a good fit for this?